lib/ruby-debug-ide/xml_printer.rb

```diff
@@ -413,7 +415,7 @@ def compact_array_str(value)
       compact = if (defined?(JRUBY_VERSION) || ENV['DEBUGGER_MEMORY_LIMIT'].to_i <= 0)
                   slice.inspect
                 else  
-                  compact = inspect_with_allocation_control(slice, ENV['DEBUGGER_MEMORY_LIMIT'].to_i)
+                  compact = inspect_with_allocation_control(slice, ENV['DEBUGGER_MEMORY_LIMIT'].to_i, value.object_id.to_s)
```

Comment:
I am not sure `object_id` us really useful for logging purposes: it does not tell anything about the object. I think the previous message already a good start.